table/block_based/full_filter_block.h

```diff
@@ -141,6 +142,8 @@ class FullFilterBlockReader
                 const SliceTransform* prefix_extractor,
                 BlockCacheLookupContext* lookup_context,
                 Env::IOPriority rate_limiter_priority) const;
+
+  const ReadOptions read_options_;
```

Comment:
I think it would make more sense to use the `ReadOptions` associated with the user read that triggers the `GetOrReadFilterBlock()`. Plus it would save `sizeof(ReadOptions)` bytes of memory per open table. Same for `PartitionedFilterBlockReader`.